nautobot_ssot/contrib.py

```diff
@@ -91,7 +91,8 @@ def _load_single_object(self, database_object, diffsync_model, parameter_names):
             metadata_for_this_field = getattr(type_hints[parameter_name], "__metadata__", [])
             for metadata in metadata_for_this_field:
                 if isinstance(metadata, CustomFieldAnnotation):
-                    parameters[parameter_name] = database_object.cf[metadata.name]
+                    if metadata.name in database_object.cf:
+                        parameters[parameter_name] = database_object.cf[metadata.name]
```

Comment:
Just a note to myself for later review:

If the pydantic model field that corresponds to this parameter doesn't have a default and we don't hit this `if` here, this will likely result in an error further down:

```
        try:
            diffsync_model = diffsync_model(**parameters)
        except pydantic.ValidationError as error:
            raise ValueError(f"Parameters: {parameters}") from error
```

Probably we want to handle this explicitly and provide an error message if it is the case.